lib/utils/fs.go

```diff
@@ -152,3 +152,38 @@ func StatDir(path string) (os.FileInfo, error) {
 	}
 	return fi, nil
 }
+
+type FilePermissions struct {
+	WorldExecute bool
+	WorldWrite   bool
+	WorldRead    bool
+
+	GroupExecute bool
+	GroupWrite   bool
+	GroupRead    bool
+
+	OwnerExecute bool
+	OwnerWrite   bool
+	OwnerRead    bool
+}
+
+func FilePerms(path string) (FilePermissions, error) {
```

Comment:
Do we really need that utility to check the file permissions? If you think that `(1<<2)` is too cryptic you can always add a comment or create `const` to make code more readable:

```go
const worldWritableBits = 0004

perm := fi.Mode().Perm()
isWorldWritable := perm&worldWritableBits
```

Reply:
Hmm, it's certainly not a necessity.  I originally wrote it with the bit shifts directly in there, but was worried that it was a little cryptic. I can revert back if you think that pulling this out into a utility isn't providing much value.

Reply:
My main concern here is readability when you want to check more than one permission. I think most people know what `0755` or `0644` are and it's easy to check the permission in one line. Doing the same thing here would require to check each individual bit, which is more verbose. Also, Go standard library follows the same Posix pattern: https://pkg.go.dev/os#FileMode

@r0mant Any opinions?

Reply:
We already have a few file permissions as constants, so there's somewhat of a precedent. https://github.com/gravitational/teleport/blob/38b24f68032d76768dca5a7c98758e48dce282dd/constants.go#L312-L318